### PR TITLE
Hosted Gallery - fix landscape ratio & add onward journey

### PR DESCRIPTION
**Pull request by lps88**
## What does this change?
- Landscape images are now always 5:3
- header/footer items align to image width
- New onward journey options at the end of the gallery.
## Screenshots

image
## Request for comment

@Calanthe @kelvin-chappell

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js [hunk 1]
```diff
@@ -357,23 +364,54 @@ define([
...
+            $footer = this.$captionContainer,
+            $progress = this.$progress,
+            imgRatio = 5/3,
+            imageWidth = width,
+            leftRight = 0;
+        if(imgRatio < width/height) {
+            imageWidth = height * imgRatio;
+            leftRight = (width - imageWidth) / 2 + 'px';
+        }
+        $header.css('width', imageWidth);
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: Wrap those css changes in fastdom.write

File: static/src/stylesheets/module/commercial/glabs/_hosted-gallery.scss [hunk 2]
```diff
@@ -62,10 +66,15 @@
...
         position: absolute;
         overflow: hidden;
+        width: 100%;
+        height: 100%;
     }
     .u-responsive-ratio {
         position: static;
     }
+    img {
+        object-fit: cover;
```
Comment (Calanthe) [inline static/src/stylesheets/module/commercial/glabs/_hosted-gallery.scss]: How does the gallery look and work in IE? The `object-fit: cover` is not supported there.

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js [hunk 3]
```diff
@@ -156,53 +159,60 @@ define([
...
+        if(imgRatio > width/height) {
+            // portrait screens
+            imageHeight = width / imgRatio;
+            topBottom = (height - imageHeight) / 2;
+        } else {
+            // landscape screens
+            imageWidth = height * imgRatio;
+            leftRight = (width - imageWidth) / 2;
+        }
+        $sizer.css('width', imageWidth);
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: Use fastom.write

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js [hunk 4]
```diff
@@ -44,6 +44,7 @@ define([
 
         // ELEMENT BINDINGS
         this.$galleryEl = $('.js-hosted-gallery-container');
+        this.$header = $('.hosted__headerwrap');
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: Use the js hook `js-hosted__headerwrap`

File: commercial/app/views/hosted/guardianHostedGallery.scala.html [hunk 5]
```diff
@@ -17,7 +17,9 @@ <h3 class="hosted-gallery__sub-heading">{page.standfirst}</h3>
                             </div>
                         }.toString)
                         @for(i <- page.images.indices.tail) {
-                            @guardianHostedGalleryImage(page.images(i), if (i == page.ctaIndex) {
+                            @guardianHostedGalleryImage(page.images(i), if (i == page.images.length - 1) {
+                                guardianHostedGalleryOj(page.ctaText, page.ctaLink, page.ctaButtonText, page.nextGalleries).toString
+                            } else  if (i == page.ctaIndex.getOrElse(-1)) {
```
Comment (kelvin-chappell) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: could be 
`} else if (page.ctaIndex.contains(i)) {`

File: commercial/app/views/hosted/guardianHostedGallery.scala.html [hunk 6]
```diff
@@ -42,72 +47,72 @@ <h3 class="hosted-gallery__sub-heading">{page.standfirst}</h3>
...
+                        </div>
+                    </header>
+                    <div class="hosted-gallery__captions">
+                        @for(i <- page.images.indices) {
+                            <div class="hosted-gallery__caption js-hosted-gallery-caption @{if (i == 0) "current-caption" else ""}">
+                                @fragments.inlineSvg("camera", "icon", List("inline-camera inline-icon "))
+                                <span>@page.images(i).title</span>
+                                <div class="hosted-gallery__caption-text">
+                                    @page.images(i).caption
+                                    @{if (page.images(i).caption.length > 0 && page.images(i).credit.length > 0) <br/>}
```
Comment (kelvin-chappell) [inline commercial/app/views/hosted/guardianHostedGallery.scala.html]: Could be
`@if (page.images(i).caption.nonEmpty && page.images(i).credit.nonEmpty) {<br/>}`

File: static/src/stylesheets/module/commercial/glabs/_hosted-gallery.scss (hunk 2 above)
Comment (lps88) [inline static/src/stylesheets/module/commercial/glabs/_hosted-gallery.scss]: Ben is cropping the images in the grid, so they will hopefully be the right size and won't get stretched! Just have to update the URLs next

**Comment by kelvin-chappell** [comment]
Looks good.  Just a couple of trivial comments from me.

👍  when @Calanthe is happy

File: static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js [hunk 7]
```diff
@@ -357,23 +372,59 @@ define([
...
-                    }
+                'resize': function(){
+                    this.onResize();
                 }
             }
         }
     };
 
+    HostedGallery.prototype.onResize = function () {
+        this.swipeContainerWidth = this.$galleryEl.dim().width;
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js]: I think that this should be in fastdom.read

**Comment by Calanthe** [comment]
Other than that 👍

**Pull request merged**